src/main/java/frc/robot/subsystems/AlgaeArm.java

```diff
@@ -0,0 +1,52 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class AlgaeArm extends SubsystemBase {
+    private final DoubleSolenoid solenoid1;
+    private final DoubleSolenoid solenoid2;
+    private final DigitalInput switchTop;
+    private final DigitalInput switchBottom;
+
+    public AlgaeArm(){
+        this.solenoid1 = new DoubleSolenoid(PneumaticsModuleType.REVPH, RobotMap.ALGEA_ARM_FORWARD_PISTON_FORWARD_CHANNEL, RobotMap.ALGEA_ARM_FORWARD_PISTON_REVERSE_CHANI);
+        this.solenoid2 = new DoubleSolenoid(PneumaticsModuleType.REVPH, RobotMap.ALGEA_ARM_BACKWARD_PISTON_FORWARD_CHANNEL, RobotMap.ALGEA_ARM_BACKWARD_PISTON_REVERSE_CHANI);
+        this.switchTop = new DigitalInput(RobotMap.ALGAE_ARM_SWITCH_TOP);
+        this.switchBottom = new DigitalInput(RobotMap.ALGAE_ARM_SWITCH_BOTTOM);
+    }
+
+    public boolean isExtended(){
+        return !(switchTop.get()) && !(switchBottom.get());
```

Comment:
no, there is a switch at the top and at the bottom. Not one switch per side.
So `isExtended` is indicated by the top switch being active, `isRetracted` is indicated by the bottom switch being active.